subsys/net/l2/openthread/openthread_utils.c

```diff
@@ -269,11 +270,22 @@ void add_ipv6_maddr_to_zephyr(struct openthread_context *context)
 							 buf, sizeof(buf))));
 		}
 
-		net_if_ipv6_maddr_add(context->iface,
+		zmaddr = net_if_ipv6_maddr_add(context->iface,
 				      (struct in6_addr *)(&maddress->mAddress));
+
+		if (zmaddr &&
+		    !(net_if_ipv6_maddr_is_joined(zmaddr) ||
+		      net_ipv6_is_addr_mcast_iface((struct in6_addr *)
+						   (&maddress->mAddress)) ||
+		      net_ipv6_is_addr_mcast_link_all_nodes((struct in6_addr *)
+							    (&maddress->mAddress)))) {
```

Comment:
nit: This line exceeds the char limit, I'd suggest to move the cast to the line below like this:
```suggestion
		      net_ipv6_is_addr_mcast_link_all_nodes(
		      		(struct in6_addr *)(&maddress->mAddress)))) {
```
This could apply for `net_ipv6_is_addr_mcast_iface` as well for consistency.